cmd/notary/keys.go

```diff
@@ -258,24 +267,43 @@ func (k *keyCommander) keysExportRoot(cmd *cobra.Command, args []string) error {
 	if err != nil {
 		return err
 	}
-	cs := cryptoservice.NewCryptoService("", ks...)
+
+	// Search for this key in all of our keystores, determine whether this key has a GUN
```

Comment:
Yeah, the flatkeystore or at least getting key info by ID will help this.  I'm looking into this and hope to open another PR that can provide this functionality.

Until then we need the loop to find the gun unless we want to change the current client API for exporting keys.  We have a similar loop in `removeKeyInteractively` that could also be improved